rtcp/src/transport_feedbacks/transport_layer_nack/mod.rs

```diff
@@ -27,40 +27,69 @@ pub struct NackPair {
 pub type RangeFn =
     Box<dyn (Fn(u16) -> Pin<Box<dyn Future<Output = bool> + Send + 'static>>) + Send + Sync>;
 
-impl NackPair {
-    /// PacketList returns a list of Nack'd packets that's referenced by a NackPair
-    pub fn packet_list(&self) -> Vec<u16> {
-        let mut out = vec![self.packet_id];
+pub struct NackIterator {
+    packet_id: u16,
+    bitfield: PacketBitmap,
+    index: Option<u16>,
+}
+
+impl Iterator for NackIterator {
+    type Item = u16;
 
-        let mut b = self.lost_packets;
-        let mut i = 0;
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.index {
+            None => {
+                self.index = Some(0);
 
-        while b != 0 {
-            if (b & (1 << i)) != 0 {
-                b &= !(1 << i);
-                out.push(self.packet_id + i + 1);
+                Some(self.packet_id)
             }
-            i += 1;
+            Some(id) if self.bitfield != 0 => {
+                let mut i = id;
+
+                while self.bitfield != 0 {
+                    if (self.bitfield & (1 << i)) != 0 {
+                        self.bitfield &= !(1 << i);
```

Comment:
This looks strange. If we have this:

```
packet_id: 123
bitfield: `00000000 00000101`
```

We expect the iterator to produce `123`, `124`, and `126`. 

But it looks like `self.bitfield` is "destroyed" on the first iteration which means the result would be `123`, `124` – and missing the last value.

Reply:
This specific line zeroes out the bit in question which ensure we eventually hit the loop condition.

Keep in mind that we are within a `next` method here, as such we only run until we can produce a value, subsequent calls to `next` should produce another value if we do indeed have more values to produce.

I'll write a test with a gap like you pointed out to verify